chef/cookbooks/monasca/recipes/master.rb

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright 2017 Fujitsu LIMITED
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitation.
+
+package "ansible"
+
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+directory "/opt/fujitsu" do
+  owner "root"
+  group "root"
+  mode "0755"
+end
+
+# TODO: use package "monasca-installer"
+remote_file "/opt/fujitsu/cmm-suse.tgz" do
+  source node[:monasca][:master][:cmm_tarball_url]
+  owner "root"
+  group "root"
+  mode "0644"
+  action :create_if_missing
+end
+
+execute "extract cmm tarball" do
+  command "tar xf cmm-suse.tgz"
+  cwd "/opt/fujitsu/"
+  not_if { Dir.exist?("/opt/fujitsu/monasca-installer") }
+  notifies :run, "execute[run ansible]", :delayed
+end
+
+template "/opt/fujitsu/monasca-installer/credentials.yml" do
+  source "credentials.yml.erb"
+  owner "root"
+  group "root"
+  mode "0644"
+  variables(
+    keystone_settings: keystone_settings
+  )
+  notifies :run, "execute[run ansible]", :delayed
+end
+
+monasca_hosts = MonascaHelper.monasca_hosts(search(:node, "roles:monasca-server"))
+
+if monasca_hosts.nil? or monasca_hosts.empty?
+  fail "no nodes with monasca-server role found"
+elsif monasca_hosts.length == 1
```

Comment:
Fail for `monasca_hosts.length == 2`
We need 1 node for single and at last 3 nodes for cluster